src/engine/PathFinding.hpp

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include <deque>
+#include <unordered_set>
+#include <vector>
+
+#include "DataObjects.hpp"
+#include "GameObject.hpp"
+#include "QuadTree.hpp"
+
+namespace admirals {
+
+class PathFinding {
+private:
+    struct Node {
+        bool visited = false;
+        float g = 0;
+        float h = 0;
+        float f = 0;
```

Comment:
Either rename these to more descriptive values or add comments to them

Reply:
The A* algorithm refers to these values as g, h, f. They do not have clear names (except h for heuristc).